### ISSUE TITLE
ember-data 2.13+ throws error when booting ember-engines

### ISSUE THREAD

**Issue by rwjblue** [emberjs/data#4968]
The deprecations added as part of https://github.com/emberjs/data/pull/4657 are written in a way that assumes the argument passed into the `instance-initializer` has an `application` property. This is true for normal applications, but is not true for engines (which still run our `instance-initializers`). 

Specifically [this line] errors (because `application.application` is `null`):

```
  deprecateOldEmberDataInitializers(application.application.constructor.initializers);
```

**Comment by rwjblue** [comment]
I'm looking at fixing now...

**Comment by rwjblue** [comment]
Submitted PR's to fix ( #4969 and #4970).

**Comment by sly7-7** [comment]
Likely fixed 😄  Thanks @rwjblue
